python/seldon_core/seldon_methods.py

```diff
@@ -314,15 +320,30 @@ def aggregate(
         if is_proto:
             features_list = []
             names_list = []
+            meta_list = []
 
             for msg in request.seldonMessages:
                 (features, meta, datadef, data_type) = extract_request_parts(msg)
                 features_list.append(features)
                 names_list.append(datadef.names)
+                meta_list.append(meta)
+
+            tags = {}
+            for meta in meta_list:
+                if (
+                    (meta is not None)
+                    and ("tags" in meta)
+                    and (meta["tags"] is not None)
+                ):
+                    tags.update(meta["tags"])
 
             client_response = client_aggregate(user_model, features_list, names_list)
             return construct_response(
```

Comment:
Based on how aggregate is handling in two different parts, it seems like it may make more sense to add the logic to aggregate the metadata in the client_transform_output, client_predict, etc, instead of the construct_response. Is there a particular reason why you chose to handle this in the construct_Response instead of the respective client_METHOD functions?

Reply:
All because `client_custom_tags` were called there: https://github.com/SeldonIO/seldon-core/blob/c682547c4e17ead1a9d96198ac520c32db441120/python/seldon_core/utils.py#L414

Reply:
Also `client_METHOD` functions where only passing `meta` through `kwargs`.